raiden_contracts/tests/test_token_network.py

```diff
@@ -317,6 +317,46 @@ def test_constructor_call(
             token_network_deposit_limit,
         ])
 
+    # failure with channel_participant_deposit_limit being zero
+    with pytest.raises(TransactionFailed):
+        get_token_network([
+            custom_token.address,
+            secret_registry_contract.address,
+            chain_id,
+            TEST_SETTLE_TIMEOUT_MIN,
+            TEST_SETTLE_TIMEOUT_MAX,
+            deprecation_executor,
+            0,
+            token_network_deposit_limit,
+        ])
+
+    # failure with both limits being zero
+    with pytest.raises(TransactionFailed):
+        get_token_network([
+            custom_token.address,
+            secret_registry_contract.address,
+            chain_id,
+            TEST_SETTLE_TIMEOUT_MIN,
+            TEST_SETTLE_TIMEOUT_MAX,
+            deprecation_executor,
+            0,
+            0,
+        ])
+
+    # failure with channel_participant_deposit_limit being bigger than
+    # token_network_deposit_limit.
+    with pytest.raises(TransactionFailed):
+        get_token_network([
+            custom_token.address,
+            secret_registry_contract.address,
+            chain_id,
+            TEST_SETTLE_TIMEOUT_MIN,
+            TEST_SETTLE_TIMEOUT_MAX,
+            deprecation_executor,
+            token_network_deposit_limit,
+            channel_participant_deposit_limit,
```

Comment:
Too bad that we can't use keyword args when calling solidity functions. This would make this much clearer.

Reply:
Actually, `web3.py` seems to support keyword args
```
token_network_registry_contract.functions.createERC20TokenNetwork(
        _token_address=custom_token.address,
        _channel_participant_deposit_limit=0,
        _token_network_deposit_limit=0,
    )
```

Reply:
I didn't realize. That's excellent to know!

Reply:
https://github.com/raiden-network/raiden-contracts/issues/672